src/include/OpenImageIO/fmath.h

```diff
@@ -2068,7 +2068,10 @@ OIIO_FORCEINLINE OIIO_HOSTDEVICE T fast_exp2 (const T& xval) {
     r = madd(x, r, kD);
     r = madd(x, r, kE);
     r = madd(x, r, one);
-    return bitcast_to_float (bitcast_to_int(r) + (m << 23));
+
```

Comment:
Suggestion to please add a comment here along the lines of:

```
  // Original code was:
  //     return bitcast_to_float (bitcast_to_int(r) + (m << 23));
  // This was producing the wrong results when <explain configuration>
  // and we found that it was fixed by using a temporary intN, as below.
  // Presumed to be an optimizer bug in <explain compiler>?
  // See PR #3804
```

Reply:
Done and squashed onto the original commit into daeda454412beaa3f305287acaad115e46f7d620